pkg/alertmanager/alertmanager.go

```diff
@@ -474,7 +476,7 @@ func buildReceiverIntegrations(nc config.Receiver, tmpl *template.Template, fire
 				return
 			}
 			n = wrapper(name, n)
-			integrations = append(integrations, notify.NewIntegration(n, rs, name, i))
+			integrations = append(integrations, notify.NewIntegration(n, rs, name, i, nc.Name))
```

Comment:
Directly parallels the prometheus code.

https://github.com/prometheus/alertmanager/blob/fdea7e731cd0b07cb2a342f182dd4d37938b69db/config/receiver/receiver.go#L52